auth-server/src/services/auth.ts

```diff
@@ -273,6 +276,35 @@ export default class AuthService {
 			throw new ForbiddenError(messages.USER_EMAIL_ALREADY_EXISTS);
 		}
 
+		let user = await getUserFromUserId(userId);
+
+		const existingUndoToken = await UndoEmailChangeToken
+			.query()
+			.where('user_id', userId)
+			.orderBy('id', 'desc')
+			.first();
+
+		if (existingUndoToken) {
+			const now = moment();
+			const last = moment(existingUndoToken.created_at);
+
+			const hours = moment.duration(last.diff(now)).asHours();
```

Comment:
This is wrong, the diff is always negative and results in users never ever be able to change their email.
Changing my mind regarding adding tests in another PR, this is actually a must. If I hadn't tested this manually I wouldn't have caught this bug, and that's bad.

```suggestion
			const hours = moment.duration(now.diff(last)).asHours();
```